crates/polars-time/src/windows/window.rs

```diff
@@ -157,14 +224,20 @@ impl<'a> BoundsIter<'a> {
                 boundary
             },
             StartBy::WindowBound => match tu {
-                TimeUnit::Nanoseconds => window.get_earliest_bounds_ns(boundary.start, tz)?,
-                TimeUnit::Microseconds => window.get_earliest_bounds_us(boundary.start, tz)?,
-                TimeUnit::Milliseconds => window.get_earliest_bounds_ms(boundary.start, tz)?,
+                TimeUnit::Nanoseconds => {
+                    window.get_earliest_bounds_ns(boundary.start, closed_window, tz)?
+                },
+                TimeUnit::Microseconds => {
+                    window.get_earliest_bounds_us(boundary.start, closed_window, tz)?
+                },
+                TimeUnit::Milliseconds => {
+                    window.get_earliest_bounds_ms(boundary.start, closed_window, tz)?
+                },
             },
             _ => {
                 {
                     #[allow(clippy::type_complexity)]
-                    let (from, to, offset): (
+                    let (from, to, offset_fn): (
```

Comment:
drive-by: renaming to `offset_fn`, partially because that what it's called in other places, and also because otherwise it risks being confused with the `offset` parameter of `group_by_dynamic`